compiler/modules/bir/bir.bal

```diff
@@ -308,12 +308,20 @@ public type ConvertToFloatInsn readonly & record {|
 
 public type OrderType UniformOrderType|OptOrderType|ArrayOrderType;
 public type UniformOrderType t:UT_FLOAT|t:UT_INT|t:UT_BOOLEAN|t:UT_STRING;
+
+const NoOrderType = false;
 public type OptOrderType readonly & record {|
-    UniformOrderType opt;
+    ArrayOrderType|UniformOrderType|NoOrderType opt;
 |};
-public type ArrayOrderType readonly & [OptOrderType];
 
-# This does ordered comparision
+public final OptOrderType nilOrderType = { opt: NoOrderType };
```

Comment:
For this an emptyTypeOrderType, the only reason they are not `const` is that jBallerina does not support it yet, so let's name them like consts in ALL_UPPER_CASE.